addons/product_images/wizard/product_get_pic_using_barcode.py

```diff
@@ -0,0 +1,115 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import json
+import re
+import requests
+from requests.exceptions import HTTPError, Timeout
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class ProductGetPicUsingBarcode(models.TransientModel):
+    """ Get product pictures using barcode on Google Image
+
+    :raises UserError: If the user didn't enter API Key or Search Engine ID in the settings
+    :raises UserError: If there isn't any product without a picture and with a barcode selected.
+    """
+    _name = 'product.images.get.pic.barcode'
+    _description = 'Get Picture with Barcode Number'
+
+    @api.model
+    def default_get(self, fields):
+        default = super().default_get(fields)
+
+        google_custom_search_cx = bool(self.env['ir.config_parameter'] \
+            .get_param('google.custom_search.cx'))
+        google_custom_search_key = bool(self.env['ir.config_parameter'] \
+            .get_param('google.custom_search.key'))
+
+        if not google_custom_search_cx or not google_custom_search_key:
+            raise UserError(_("You didn't configure properly your API Key or Search Engine ID."))
+
+        if self._context.get('active_model') == 'product.template':
+            product_ids = self.env['product.template'].browse(self._context.get('active_ids')) \
+                .product_variant_ids.filtered(lambda p: not p.image_1920 and p.barcode)
+        else:
+            product_ids = self.env['product.product'].browse(self._context.get('active_ids')) \
+                .filtered(lambda p: not p.image_1920 and p.barcode).ids
+
+        if len(product_ids) == 0:
+            raise UserError(_("There isn't any product without a picture and with a barcode."))
+        # TODO: Define a maximum number of product to process in the same batch
+        if len(product_ids) > 100:
+            raise UserError(_("You selected too many product variants at the same time. \
+Please select a maximum of 100 product variants in the same batch."))
```

Comment:
That's... a bit weird. Because this occurs after filtering (so sometimes selecting 150 variants will work and other times it won't), and because this can be called on product templates which might have hundreds of variants, in which case... it might not be clear for the user how to select "less than 100" variants?

Reply:
Hello Xavier, 

After some research about this comment, I couldn't solve entirely this problem. 

In my latest commit, you can see that I don't raise an UserError anymore and I've set the limits to 10K. If you have selected more than 10K Product Variants (PV), a message will be displayed in the wizard to inform you that you have selected too many PV and that only the 10K first PV of the list will be processed.

https://github.com/odoo/odoo/blob/8b45cce6897e058e1a2e42cb0ad324c5acb9575f/addons/product_images/wizard/product_get_pic_using_barcode.xml#L13-L15

The problem that persists is the distinction between Products and Product Variants. The way I made that works only with PV and when you select a product template, I fetch the PV linked to this template.

https://github.com/odoo/odoo/blob/8b45cce6897e058e1a2e42cb0ad324c5acb9575f/addons/product_images/wizard/product_get_pic_using_barcode.py#L54-L57

So if you select on product template, you may really have selected a hundred or more PV... One solution for that would be to only shows the action in the PV view but it would restrict the functionality to PV users only. To solve that, we may hide the action from the product template view if PV is activated, but it could be misleading to have the action in one view and not in the other.